pkg/startup/startup.go

```diff
@@ -1142,3 +1208,19 @@ func terminate() {
 	log.Warn("Terminating")
 	exitFunction(1)
 }
+
+// queryKubeadmConfigMap returns kubeadm's "kubeadm-config" config map if it exists.
+func queryKubeadmConfigMap(clientset *kubernetes.Clientset) (*kapiv1.ConfigMap, error) {
+	// clientset will only be initialized when running on k8s
+	if clientset == nil {
+		return nil, fmt.Errorf("Unable to read kubeadm config map, not running on kubernetes.")
+	}
+
+	kubeadmConfig, err := clientset.CoreV1().ConfigMaps(metav1.NamespaceSystem).Get(KubeadmConfigConfigMap, metav1.GetOptions{})
+	if err != nil {
```

Comment:
I think we need to handle `IsNotFound` explicitly here as well. Otherwise we'll log out an error on non-kubeadm systems.